### ISSUE TITLE
tests for php7

### ISSUE THREAD

**Issue by zordius** [zordius/lightncandy#181]
Need some fix for the test framework to prevent errors like this:

```
Parameter 2 to LightnCandy::prePartial() expected to be a reference, value given
```

It's caused by the `ReflectionMethod->invoke()` , it can not handle `pass by ref` .
But, if we move to `ReflectionMethod->invokeArgs()` , we still need another way to deal with input array.

**Comment by jayS-de** [comment]
Just saw that you removed php7 from your CI matrix. You could add to your .travis.yml
```
language: php
php:
  - "5.6"
  - "5.5"
  - "5.4"
  - "7.0"
  - "hhvm"

matrix:
  allow_failures:
    - php: hhvm
    - php: "7.0"

So you could let the php7 tests run, but they won't break the CI build :)

**Comment by zordius** [comment]
Good point! I will turn tests on php7 later.

**Comment by zordius** [comment]
fixed in ada730928bd68acd43b88cd752abe1cdaebf5595 (now in branch v0.89-develop)
